packages/components/src/color-palette/utils.ts

```diff
@@ -38,9 +40,10 @@ export const extractColorNameFromCurrentValue = (
 		: [ { colors: colors as ColorObject[] } ];
 	for ( const { colors: paletteColors } of colorPalettes ) {
 		for ( const { name: colorName, color: colorValue } of paletteColors ) {
-			const normalizedColorValue = currentValueIsCssVariable
-				? colorValue
-				: colord( colorValue ).toHex();
+			const normalizedColorValue =
+				currentValueIsCssVariable || currentValueIsColorMix
+					? colorValue
+					: colord( colorValue ).toHex();
```

Comment:
We should reuse the `isSimpleCSSColor` function here, something like this should work:

```diff
diff --git a/packages/components/src/color-palette/utils.ts b/packages/components/src/color-palette/utils.ts
index d5f81f3304..88e5c89dd9 100644
--- a/packages/components/src/color-palette/utils.ts
+++ b/packages/components/src/color-palette/utils.ts
@@ -17,6 +17,19 @@ import type { ColorObject, ColorPaletteProps, PaletteObject } from './types';
 
 extend( [ namesPlugin, a11yPlugin ] );
 
+/**
+ * Checks if a color value is a simple CSS color, its value doesn't need
+ * further processing to be used as a color.
+ *
+ * @param value The color value to check.
+ * @return A boolean indicating whether the color value is a simple CSS color.
+ */
+const isSimpleCSSColor = ( value: string | undefined ) => {
+	const valueIsCssVariable = /var\(/.test( value ?? '' );
+	const valueIsColorMix = /color-mix\(/.test( value ?? '' );
+	return ! valueIsCssVariable && ! valueIsColorMix;
+};
+
 export const extractColorNameFromCurrentValue = (
 	currentValue?: ColorPaletteProps[ 'value' ],
 	colors: ColorPaletteProps[ 'colors' ] = [],
@@ -26,12 +39,10 @@ export const extractColorNameFromCurrentValue = (
 		return '';
 	}
 
-	const currentValueIsCssVariable = /^var\(/.test( currentValue );
-	const currentValueIsColorMix = /color-mix\(/.test( currentValue );
-	const normalizedCurrentValue =
-		currentValueIsCssVariable || currentValueIsColorMix
-			? currentValue
-			: colord( currentValue ).toHex();
+	const currentValueIsSimpleColor = isSimpleCSSColor( currentValue );
+	const normalizedCurrentValue = currentValueIsSimpleColor
+		? colord( currentValue ).toHex()
+		: currentValue;
 
 	// Normalize format of `colors` to simplify the following loop
 	type normalizedPaletteObject = { colors: ColorObject[] };
@@ -40,10 +51,9 @@ export const extractColorNameFromCurrentValue = (
 		: [ { colors: colors as ColorObject[] } ];
 	for ( const { colors: paletteColors } of colorPalettes ) {
 		for ( const { name: colorName, color: colorValue } of paletteColors ) {
-			const normalizedColorValue =
-				currentValueIsCssVariable || currentValueIsColorMix
-					? colorValue
-					: colord( colorValue ).toHex();
+			const normalizedColorValue = currentValueIsSimpleColor
+				? colord( colorValue ).toHex()
+				: colorValue;
 
 			if ( normalizedCurrentValue === normalizedColorValue ) {
 				return colorName;
@@ -71,18 +81,6 @@ export const isMultiplePaletteArray = (
 	);
 };
 
-/**
- * Checks if a color value is a simple CSS color.
- *
- * @param value The color value to check.
- * @return A boolean indicating whether the color value is a simple CSS color.
- */
-const isSimpleCSSColor = ( value: string | undefined ): boolean => {
-	const valueIsCssVariable = /var\(/.test( value ?? '' );
-	const valueIsColorMix = /color-mix\(/.test( value ?? '' );
-	return ! valueIsCssVariable && ! valueIsColorMix;
-};
-
 /**
  * Transform a CSS variable used as background color into the color value itself.
  *

```